primitives/src/targeting/eval.rs

```diff
@@ -475,6 +879,61 @@ enum MathOperator {
     Min,
 }
 
+enum ComparisonOperator {
+    Gt,
+    Gte,
+    Lt,
+    Lte,
+    Equals,
+    NotEquals,
+}
+
+fn compare_numbers(lhs: Number, rhs: Number, ops: ComparisonOperator) -> Result<bool, Error> {
+    match (lhs.as_u64(), rhs.as_u64()) {
+        (Some(lhs), Some(rhs)) => handle_comparisons(lhs, rhs, ops),
+        _ => match (lhs.as_i64(), rhs.as_i64()) {
+            (Some(lhs), Some(rhs)) => handle_comparisons(lhs, rhs, ops),
+            _ => match (lhs.as_f64(), rhs.as_f64()) {
+                (Some(lhs), Some(rhs)) => handle_comparisons(lhs, rhs, ops),
+                _ => Err(Error::TypeError),
+            },
+        },
+    }
+}
+
+fn handle_comparisons<T: PartialOrd>(
+    lhs: T,
+    rhs: T,
+    ops: ComparisonOperator,
+) -> Result<bool, Error> {
+    match ops {
+        ComparisonOperator::Lt => {
+            let is_lt = lhs < rhs;
```

Comment:
Note: you could actually invoke the functions of PartialEq/PartialOrd directly.
I.e. `lhs.lt(rhs)`